driver-core/src/test/unit/com/mongodb/internal/connection/IdHoldingBsonWriterSpecification.groovy

```diff
@@ -88,6 +88,32 @@ class IdHoldingBsonWriterSpecification extends Specification {
         encodedDocument == document
     }
 
+    def 'serialize _id documents containing arrays'() {
+        def bsonBinaryWriter = new BsonBinaryWriter(new BasicOutputBuffer())
+        def idTrackingBsonWriter = new IdHoldingBsonWriter(bsonBinaryWriter)
+        BsonDocument document = BsonDocument.parse(json)
+
+        when:
+        new BsonDocumentCodec().encode(idTrackingBsonWriter, document, EncoderContext.builder()
+                .isEncodingCollectibleDocument(true).build())
+        def encodedDocument = getEncodedDocument(bsonBinaryWriter.getBsonOutput())
+
+        then:
+        encodedDocument == document
+
+        where:
+        json << ['{"_id": {"a": []}, "b": 123}',
```

Comment:
Did all of these fail before the fix?

Reply:
No I added extra for regression purposes.

These tests failed (where the array is a sub field of a document of the _id):

```
{"_id": {"a": [1, 2]}, "b": 123},
{"_id": {"a": [[[[1]]]]}, "b": 123},
{"_id": {"a": []}, "b": 123},
{"_id": {"a": [{"a": [1, 2]}]}, "b": 123},